### ISSUE TITLE
Retry if file was not found

### ISSUE THREAD

**Issue by chgarling** [HeavyHorst/remco#29]
Today we had an issue with Apache Rewrites that were not updated, because the file was not available in that moment remco tried to use it. Is it possible to configure a retry for such cases?

time="2019-07-16T10:39:48Z" level=debug msg="retrieving keys" backend=file key_prefix= prefix="remco_linux[1]" resource=apache-rewrites
time="2019-07-16T10:39:48Z" level=error msg="setVars failed: getValues failed: open /usr/local/rewrites/rewrites.json: no such file or directory" backend=file prefix="remco_linux[1]" resource=apache-rewrites
time="2019-07-16T10:39:48Z" level=error msg="no such file or directory" backend=file prefix="remco_linux[1]" resource=apache-rewrites

**Comment by HeavyHorst** [comment]
Which version of remco do you use?
The current version should already try to recreate the template.

```
[0000]  INFO remco[6763]: set file path backend=file filepath=/home/rkaufmann/Downloads/test.yml
[0000] DEBUG remco[6763]: retrieving keys backend=file key_prefix= resource=test
[0000] ERROR remco[6763]: setVars failed: getValues failed: open /home/rkaufmann/Downloads/test.yml: no such file or directory backend=file resource=test
[0000] ERROR remco[6763]: not all templates could be rendered, trying again after 27 seconds resource=test
```

This even works in "onetime" mode.

**Comment by gsbab** [comment]
We have updated to version 0.11.0 now.

In version 0.11.0, remco tries to read the deleted file in a loop, but does not re-generate the configuration after the file re-appears.
Only when the content of the file changes again does remco read in the file and generate the configuration.

With a "touch -m" we would have a workaround.

**Comment by HeavyHorst** [comment]
Ah I see.
I'll provide a fix soon.

In the meantime you could set an interval in addition to the watch.


```
  [backend]
    [backend.file]
      filepath = "/etc/remco/test.yml"
      watch    = true
      interval = 10
      keys     = ["/"]
```

```
[0000]  INFO remco[10865]: loading resource configuration path=/etc/remco/resource.d/test.toml
[0000]  INFO remco[10865]: set file path backend=file filepath=/etc/remco/test.yml
[0000] DEBUG remco[10865]: retrieving keys backend=file key_prefix= resource=haproxy
[0000] DEBUG remco[10865]: compiling source template resource=haproxy template=/etc/remco/templates/test.cfg
[0000] DEBUG remco[10865]: comparing staged and dest config files dest=/home/rkaufmann/Downloads/remcotest resource=haproxy staged=.remcotest202482998
[0000] DEBUG remco[10865]: target config in sync config=/home/rkaufmann/Downloads/remcotest resource=haproxy
[0000] DEBUG remco[10865]: Running "echo 1" resource=haproxy
[0000] DEBUG remco[10865]: "1\n" resource=haproxy
[0010] DEBUG remco[10865]: retrieving keys backend=file key_prefix= resource=haproxy
[0010] DEBUG remco[10865]: compiling source template resource=haproxy template=/etc/remco/templates/test.cfg
[0010] DEBUG remco[10865]: comparing staged and dest config files dest=/home/rkaufmann/Downloads/remcotest resource=haproxy staged=.remcotest726532381
[0010] DEBUG remco[10865]: target config in sync config=/home/rkaufmann/Downloads/remcotest resource=haproxy
[0012] DEBUG remco[10865]: retrieving keys backend=file key_prefix= resource=haproxy
[0012] ERROR remco[10865]: setVars failed: getValues failed: open /etc/remco/test.yml: no such file or directory backend=file resource=haproxy
[0012] ERROR remco[10865]: no such file or directory backend=file resource=haproxy
[0014] ERROR remco[10865]: no such file or directory backend=file resource=haproxy
[0016] ERROR remco[10865]: no such file or directory backend=file resource=haproxy
[0018] ERROR remco[10865]: no such file or directory backend=file resource=haproxy
[0020] DEBUG remco[10865]: retrieving keys backend=file key_prefix= resource=haproxy
[0020] ERROR remco[10865]: setVars failed: getValues failed: open /etc/remco/test.yml: no such file or directory backend=file resource=haproxy
[0030] DEBUG remco[10865]: retrieving keys backend=file key_prefix= resource=haproxy
[0030] DEBUG remco[10865]: compiling source template resource=haproxy template=/etc/remco/templates/test.cfg
[0030] DEBUG remco[10865]: comparing staged and dest config files dest=/home/rkaufmann/Downloads/remcotest resource=haproxy staged=.remcotest887355352
[0030] DEBUG remco[10865]: target config in sync config=/home/rkaufmann/Downloads/remcotest resource=haproxy
^C[0032]  INFO remco[10865]: Captured interrupt. Exiting...
[0032] DEBUG remco[10865]: closing client connection backend=file resource=haproxy

```

**Comment by chgarling** [comment]
Hi @HeavyHorst,
any news regarding the fix?

**Comment by HeavyHorst** [comment]
I'm on vacation right now and will provide a fix next week.

chgarling <notifications@github.com> schrieb am Di., 23. Juli 2019, 14:59:

**Comment by HeavyHorst** [comment]
I've published a new release (0.11.1) with the fix included. Thx!
